cmd/publicize/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"k8s.io/test-infra/prow/githubeventserver"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+
+	"k8s.io/test-infra/pkg/flagutil"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type Config struct {
+	Bonds []Bond `json:"bonds,omitempty"`
+}
+
+func (c *Config) validate() error {
+	var errs []error
+	for index, bond := range c.Bonds {
+		if err := bond.validate(); err != nil {
+			errs = append(errs, fmt.Errorf("bond[%d]: %w", index, err))
+		}
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+// bond holds the information of which repos are bonded to each other.
+type Bond struct {
```

Comment:
Why is this a `From/To` struct and not a map? We should get a lot of validation for free.

Reply:
To make the configuration file more readable and more understandable. Otherwise, it would look like

```yaml
bonds:
-  openshift-priv/origin:
     openshift/origin
....
...
```

Reply:
Well it would be:

```
bonds:
- openshift-priv/origin: openshift/origin
```

But right, I guess that does not communicate the direction that well. /shrug

Reply:
A list of structs enables us to add in configuration options on a specific bond in the future if we need to, but yes this would be simpler as a mapping